data_steward/validation/app_errors.py

```diff
@@ -56,6 +56,7 @@ class BucketNotSet(RuntimeError):
 
     def __init__(self, msg):
         super(BucketNotSet, self).__init__(self, msg)
+        self.message = msg
```

Comment:
I think it depends on the exception. Here, we know this exception is expected so we only need the message and don't care about other attributes. Also, it's easier to validate that the message is being printed with the correct params. 
If we run into a non-custom exception, we will need more information so it can help us diagnose the problem, and we'll need more info than just the message so `f'{exc}'` or possibly `f'{repr(exc)}'` should be used.

Reply:
I'm not arguing to remove this, but I don't have the context to understand it.

Like when we had the BucketDoesNotExist exception, you attached a bucket attribute.  "It didn't exist--Well, _which_ bucket didn't exist".

but you say, "Also, it's easier to validate that the message is being printed with the correct params." -- so would this be used in testing only?

One might make the case that logging in combination with a stack trace could provide enough information (just log and `raise exc`).  But that's an opinion and style (and verbose logs).

Reply:
I agree stack trace and raise are helpful for unexpected exceptions, where we need all that information. But we don't care about that information here, since we are catching a known/expected exception. We just need to catch it, print it and move on. The message attribute helps in the printing. Ideally, all of our custom exceptions should have the message attribute since it is [standard in the docs](https://docs.python.org/3.8/tutorial/errors.html#user-defined-exceptions). The testing is just an added benefit but not the main purpose.

Reply:
gold star ⭐️ for the reference.